contracts/ChainlinkPriceFeedV3.sol

```diff
@@ -42,14 +42,15 @@ contract ChainlinkPriceFeedV3 is IChainlinkPriceFeedV3, IPriceFeedUpdate, BlockC
     }
 
     /// @notice anyone can help with updating
-    /// @dev keep this function for PriceFeedUpdater for updating, since multiple updates
-    ///      with the same timestamp will get reverted in CumulativeTwap._update()
+    /// @dev this function is used by PriceFeedUpdater for _lastValidPrice and twap updating,
+    ///      The keeper can invoke callstatic on this function to check if _lastValidPrice needs to be updated.
+    ///      If _lastValidPrice is updated, then update twap with _lastValidPrice and _lastValidTimestamp.
     function update() external override {
-        _cachePrice();
-
-        (bool isUpdated, ) = _cacheTwap(0, _lastValidPrice, _lastValidTimestamp);
+        bool isUpdated = _cachePrice();
         // CPF_NU: not updated
         require(isUpdated, "CPF_NU");
+
+        _cacheTwap(0, _lastValidPrice, _lastValidTimestamp);
```

Comment:
should we note that "instead of directly calling `_update`, by ussing `_cacheTwap`, we can cache twap and update obsercation as well"?